packages/netlify-cms-backend-gitlab/src/implementation.js

```diff
@@ -68,21 +68,31 @@ export default class GitLab {
   }
 
   entriesByFolder(collection, extension) {
-    return this.api.listFiles(collection.get('folder')).then(({ files, cursor }) =>
-      this.fetchFiles(files.filter(file => file.name.endsWith('.' + extension))).then(
-        fetchedFiles => {
+    return this.api
+      .listFiles(collection.get('folder'))
+      .then(({ files, cursor }) => ({
+        files: files.filter(file => file.name.endsWith('.' + extension)),
+        cursor,
+      }))
+      .then(({ files, cursor }) =>
+        this.fetchFiles(files).then(fetchedFiles => {
           const returnedFiles = fetchedFiles;
           returnedFiles[CURSOR_COMPATIBILITY_SYMBOL] = cursor;
           return returnedFiles;
-        },
-      ),
-    );
+        }),
+      );
   }
 
   allEntriesByFolder(collection, extension) {
     return this.api
       .listAllFiles(collection.get('folder'))
-      .then(files => this.fetchFiles(files.filter(file => file.name.endsWith('.' + extension))));
+      .then(files => files.filter(file => file.name.endsWith('.' + extension)))
+      .then(files =>
+        this.fetchFiles(files).then(fetchedFiles => {
+          const returnedFiles = fetchedFiles;
```

Comment:
I wasn't sure whether it's intended to not use the `CURSOR_COMPATIBILITY_SYMBOL` in `allEntriesByFolder` (see `entriesByFolder` above),

Reply:
`allEntriesByFolder` does not use cursors at all - it returns every entry, so there's no pagination to track.